### ISSUE TITLE
Why is it impossible to double wrap?

### ISSUE THREAD

**Issue by haudren** [openai/gym#966]
I am trying to run a "new-style" gym, i.e. one that returns a dictionary of `observation, desired_goal, achieved_goal` in the `ddpg` baseline. If I attempt to use the `evaluation` option, the program crashes because it's trying to wrap the original environment first in `FlattenDictWrapper` and then in `Monitor`.

Why is it impossible to wrap an environment in multiple wrappers ? In this case, it would be very useful to both flatten and monitor.

**Comment by mpSchrader** [comment]
Hi @haudren ,
I think it is possible to use multiple wrappers. E.g. check out: https://github.com/dgriff777/rl_a3c_pytorch/blob/master/environment.py
This guy uses about 5 wrappers. ;-)

**Comment by haudren** [comment]
You're right, it's actually impossible to wrap twice with the same wrapper an environment, not double-wrapping. It would help if the message was more informative (Basically, I had a bug and was flattening my GoalEnv twice).
